code/game/objects/structures/coatrack.dm

```diff
@@ -0,0 +1,96 @@
+/obj/structure/coatrack
+	name = "Coat Rack"
+	desc = "For a detective to hang his coat and hat."
+	icon = 'icons/obj/coatrack.dmi'
+	icon_state = "coatrack0"
+	density = 1
+	anchored = 0
+	pressure_resistance = ONE_ATMOSPHERE*5
+	flags = FPRINT
+	autoignition_temperature = AUTOIGNITION_WOOD
+	fire_fuel = 3
+	var/obj/item/clothing/suit/storage/det_suit/suit = null
+	var/obj/item/clothing/head/det_hat/hat = null
+
+/obj/structure/coatrack/attack_hand(mob/user)
+	if(suit)
+		user << "<span class='notice'>You pick up the [suit] from the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		suit.loc = get_turf(src)
+		if(!user.get_active_hand())
+			user.put_in_hands(suit)
+		suit = null
+		update_icon()
+		return
+
+	if(hat)
+		user << "<span class='notice'>You pick up the [hat] from the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		hat.loc = get_turf(src)
+		if(!user.get_active_hand())
+			user.put_in_hands(hat)
+		hat = null
+		update_icon()
+		return
+
+/obj/structure/coatrack/attackby(obj/item/clothing/C, mob/user)
+	if (istype(C, /obj/item/clothing/suit/storage/det_suit) && !suit)
+		user << "<span class='notice'>You place your [C] on the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		user.drop_item()
+		C.loc = src
+		suit = C
+		update_icon()
+
+	else if (istype(C, /obj/item/clothing/head/det_hat) && !hat)
+		user << "<span class='notice'>You place your [C] on the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		user.drop_item()
```

Comment:
For future reference, drop_item takes the argument of the loc you want to drop to. So you could write user.drop_item(src).